src/modules/universal-discovery/components/select-content-button/select.content.button.component.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import './css/select.content.button.component.css';
+
+export default class SelectContentButtonComponent extends Component {
+    constructor(props) {
+        super(props);
+
+        this.toggleEnabledState = this.toggleEnabledState.bind(this);
+        this.handleSelect = this.handleSelect.bind(this);
+        this.handleUnselect = this.handleUnselect.bind(this);
+
+        this.state = {
+            selectContentEnabled: true,
+        };
+    }
+
+    componentDidMount() {
+        this.checkCanSelectContent();
+    }
+
+    componentDidUpdate() {
+        this.checkCanSelectContent();
+    }
+
+    checkCanSelectContent() {
+        const { location, canSelectContent } = this.props;
+
+        canSelectContent(location, this.toggleEnabledState);
+    }
+
+    toggleEnabledState(selectContentEnabled) {
+        if (this.state.selectContentEnabled === selectContentEnabled) {
+            return;
+        }
+
+        this.setState((state) => ({ ...state, selectContentEnabled }));
+    }
+
+    handleSelect(event) {
+        event.stopPropagation();
```

Comment:
Why not `preventDefault()`?

Reply:
Because I don't want to bubble this event to the parent.